### PR TITLE
OpenDoor laser rotation, refactor State to smach.State

### PR DESCRIPTION
**Pull request by MatthijsBurgh**

### CODE REVIEW

**Comment by MatthijsBurgh** [comment]
Uses already the new API of the tf_server, so first implement #464

**Comment by MatthijsBurgh** [comment]
I will refactor this to work with the old tf API. Do not merge at the moment

**Comment by MatthijsBurgh** [comment]
Working with old tf API. Please review @Rayman @LoyVanBeek

File: robot_smach_states/src/robot_smach_states/startup.py [hunk 1]
```diff
@@ -1,148 +1,201 @@
...
+    def execute(self, userdata=None):
         rospy.loginfo("Waiting for door...")
-        self.laser_sub = rospy.Subscriber("/"+robot.robot_name+"/base_laser/scan", LaserScan, self.process_scan)
-
-        opened_before_timout = self.door_open.wait(timeout)
+        r = Quaternion()
+        _, (r.x, r.y, r.z, r.w) = self._robot.tf_listener.lookupTransform(self._robot.base_link_frame,
+                                                                          self._robot.robot_name + '/base_laser')
+        laser_rotation = quaternionMsgToKdlRotation(r)
+        self.laser_upside_down = -math.copysign(1, math.cos(laser_rotation.GetRPY()[0]))  # -1 normal, 1 upside down
```
Comment (LoyVanBeek) [inline robot_smach_states/src/robot_smach_states/startup.py]: What is this needed for?

Comment (MatthijsBurgh) [reply robot_smach_states/src/robot_smach_states/startup.py]: In case a laser is ever not going to be position straight forward

Comment (LoyVanBeek) [reply robot_smach_states/src/robot_smach_states/startup.py]: Aha. I didn't spot the use of the variable but that's in the callback ```process_scan```

**Pull request merged**